### PR TITLE
[SPARK-40699][DOCS] Supplement undocumented yarn configurations in documentation

### PR DESCRIPTION
**Pull request by dcoliversun**
### What changes were proposed in this pull request?

This PR aims to supplement undocumented yarn configuration in documentation.

### Why are the changes needed?

Help users to confirm yarn configurations through documentation instead of code.

### Does this PR introduce _any_ user-facing change?

Yes, more configurations in documentation.

### How was this patch tested?

Pass the GA.

### CODE REVIEW

File: docs/running-on-yarn.md [hunk 1]
```diff
@@ -486,6 +486,20 @@ To use a custom metrics.properties for the application master and executors, upd
   </td>
   <td>3.3.0</td>
 </tr>
+<tr>
+  <td><code>spark.yarn.am.tokenConfRegex</code></td>
```
Comment (dcoliversun) [inline docs/running-on-yarn.md:490]: https://github.com/apache/spark/blob/c4cd193b4d43d951255d9914f9b6f099d140fe1c/resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/config.scala#L81-L96

File: docs/running-on-yarn.md [hunk 2]
```diff
@@ -632,6 +646,33 @@ To use a custom metrics.properties for the application master and executors, upd
   </td>
   <td>0.9.0</td>
 </tr>
+<tr>
+  <td><code>spark.yarn.clientLaunchMonitorInterval</code></td>
```
Comment (dcoliversun) [inline docs/running-on-yarn.md:650]: https://github.com/apache/spark/blob/c4cd193b4d43d951255d9914f9b6f099d140fe1c/resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/config.scala#L228-L233

File: docs/running-on-yarn.md [hunk 3]
```diff
@@ -632,6 +646,33 @@ To use a custom metrics.properties for the application master and executors, upd
...
+  <td><code>1s</code></td>
+  <td>
+    Interval between requests for status the client mode AM when starting the app.
+  </td>
+  <td>2.3.0</td>
+</tr>
+<tr>
+  <td><code>spark.yarn.includeDriverLogsLink</code></td>
```
Comment (dcoliversun) [inline docs/running-on-yarn.md:658]: https://github.com/apache/spark/blob/c4cd193b4d43d951255d9914f9b6f099d140fe1c/resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/config.scala#L235-L242

File: docs/running-on-yarn.md [hunk 4]
```diff
@@ -632,6 +646,33 @@ To use a custom metrics.properties for the application master and executors, upd
...
+  <td><code>false</code></td>
+  <td>
+    In cluster mode, whether the client application report includes links to the driver 
+    container's logs. This requires polling the ResourceManager's REST API, so it 
+    places some additional load on the RM.
+  </td>
+  <td>3.1.0</td>
+</tr>
+<tr>
+  <td><code>spark.yarn.unmanagedAM.enabled</code></td>
```
Comment (dcoliversun) [inline docs/running-on-yarn.md:668]: https://github.com/apache/spark/blob/c4cd193b4d43d951255d9914f9b6f099d140fe1c/resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/config.scala#L346-L351

**Review by dcoliversun — COMMENTED** [review]
cc @HyukjinKwon @dongjoon-hyun 
It would be good if you have a time to review this PR

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**